### PR TITLE
Add trigger rule tooltip

### LINKED ISSUE #21201: Add Trigger Rule Display to Graph View
### Description

This feature would introduce some visual addition(s) (e.g. tooltip) to the Graph View to display the trigger rule between tasks. 

### Use case/motivation

This would add more detail to Graph View, providing more information visually about the relationships between upstream and downstream tasks.

### Related issues

https://github.com/apache/airflow/issues/19939

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by adamfrly**
closes: #21201
related: #19939

This adds the Trigger Rule to the tooltip of operators in the graph view. It provides more detail about the relationships between up and downstream tasks in a DAG. 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by bbovenzi** [comment]
There are a few linting issues. I'd run `yarn && yarn lint` in the `/airflow/www` directory. Otherwise, looking good.

File: airflow/www/static/js/task_instances.js [hunk 1]
```diff
@@ -98,6 +98,7 @@ export default function tiTooltip(ti, { includeTryNumber = false } = {}) {
   if (ti.operator !== undefined) {
     tt += `Operator: ${escapeHtml(ti.operator)}<br>`;
   }
+  tt += `Trigger Rule: ${task.trigger_rule}<br>`;
```
Comment (bbovenzi) [inline airflow/www/static/js/task_instances.js]: Wrap this in an if statement checking that task and trigger_rule are defined.

**Review by bbovenzi — COMMENTED** [review]
Let's double check everywhere we call `tiTooltip`. I think we'll get an error in at least gantt.js, maybe other spots.

Also, care to include a screenshot of the updated tooltip so other reviewers can see what the UI change is?

**Comment by adamfrly** [comment]
Attached is a screenshot of the change to the Graph view tooltip. Note that "Trigger Rule" has been added just below the Operator indicator.
tooltip_UI_change

File: airflow/www/static/js/task_instances.js (hunk 1 above)
Comment (adamfrly) [reply airflow/www/static/js/task_instances.js]: An if statement was added in commit 3412688

**Comment by adamfrly** [comment]
> Let's double check everywhere we call `tiTooltip`. I think we'll get an error in at least gantt.js, maybe other spots.
> 
> Also, care to include a screenshot of the updated tooltip so other reviewers can see what the UI change is?

`tiTooltip` is used in gantt.js, graph.js, and dags.js. In gantt.js `null` was passed as the second argument to `tiTooltip` so that arguments are now passed properly. In the other two files however, the extra `task` argument doesn't look like it will cause any problems

File: airflow/www/static/js/task_instances.js [hunk 2]
```diff
@@ -98,7 +98,9 @@ export default function tiTooltip(ti, { includeTryNumber = false } = {}) {
   if (ti.operator !== undefined) {
     tt += `Operator: ${escapeHtml(ti.operator)}<br>`;
   }
-
+  if (task.trigger_rule !== undefined) {
```
Comment (bbovenzi) [inline airflow/www/static/js/task_instances.js]:
```suggestion
  if (task && task.trigger_rule) {
```
One last change.

**Review by bbovenzi — APPROVED** [review]
Nice work! Thanks for putting up with all my comments.

**Comment by adamfrly** [comment]
This was my first ever open source contribution so I really appreciated the comments. Thanks for making the process enjoyable!

**Pull request merged**